preprocessing/nextclade/src/loculus_preprocessing/processing_functions.py

```diff
@@ -198,6 +201,163 @@ def check_date(
                 ],
             )
 
+    @staticmethod
+    def parse_date_into_range(
+        input_data: InputMetadata,
+        output_field: str,
+        args: FunctionArgs = None,  # args is essential - even if Pylance says it's not used
+    ) -> ProcessingResult:
+        """Parse date string into a range, return based on FunctionArgs
```

Comment:
I think the supported date range format is pretty essential info that I'd like to see in the docs here.